### ISSUE TITLE
Event Hubs Troubleshooting Guide

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-go#17459]
# Summary

In order to provide guidance for customers encountering difficulties with the client library, and support representatives working customer issues, a troubleshooting guide is needed.  The guide should be focused on common scenarios for the client library and not attempt to address service-specific issues.

# Scope

- Wait for the forthcoming [Java troubleshooting guide](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/TROUBLESHOOTING.md) to be completed and made available.  _(target date is April 25, 2022)_

- Review email, issues, and the Java guide to identify any recurring topics.

- Address common scenarios, such as:
    - Event Hubs exceptions that can be returned, and guidance for them
    - Connectivity issues for enterprise environments.  _(try the web sockets transport)_
    - Common event processor scenarios (too many partitions for the machine,  load balancing can't keep up, etc)
    - Consider discussing specific sets of log events to capture for certain scenarios

- Write the guide in markdown format, intended for `/sdk/messaging/azeventhubs/TROUBLESHOOTING.md`

- Add stub guides in the root of each package pointing back at the main troubleshooting guide.  _(see KeyVault for inspiration)_

# Success Criteria

- The guide has been created and added to the repository.

# References and Resources

- [Azure.Identity Troubleshooting Guide](https://github.com/schaabs/azure-sdk-for-net/blob/id-troublehooting-update/sdk/identity/Azure.Identity/troubleshooting.md)
- [Troubleshooting Section of the README](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/eventhub/Azure.Messaging.EventHubs#troubleshooting)
- [Event Hubs Exceptions (legacy)](https://docs.microsoft.com/azure/event-hubs/event-hubs-messaging-exceptions)
- [Troubleshooting Event Hubs Connectivity](https://docs.microsoft.com/azure/event-hubs/troubleshooting-guide)

**Comment by jsquire** [comment]
//fyi: @richardpark-msft

**Comment by richardpark-msft** [comment]
This is a good thing to have - I'd prioritize Service Bus over Event Hubs, primarily because we're releasing an SB GA and haven't committed to a track 2 verson of Event Hubs yet.

**Comment by jsquire** [comment]
Agreed on prioritization; Service Bus is part of the next wave of TSGs.  There will be an equivalent issue filed once those efforts kick off.
